src/app.js

```diff
@@ -43,44 +44,89 @@ var css = csjs`
     position           : relative;
     width              : 100vw;
     height             : 100vh;
+    overflow           : hidden;
   }
   .centerpanel         {
     display            : flex;
     flex-direction     : column;
     position           : absolute;
     top                : 0;
-    left               : 200px;
-    right              : 0;
     bottom             : 0;
-  }
-  .tabsbar             {
     overflow           : hidden;
   }
   .leftpanel           {
     display            : flex;
+    flex-direction     : column;
     position           : absolute;
     top                : 0;
-    left               : 0;
-    right              : 0;
     bottom             : 0;
-    width              : 200px;
+    left               : 0;
+    overflow           : hidden;
   }
-  .dragbar2            {
-    background-color   : transparent;
+  .rightpanel          {
+    display            : flex;
+    flex-direction     : column;
     position           : absolute;
-    width              : 0.5em;
-    top                : 3em;
+    top                : 0;
+    right              : 0;
     bottom             : 0;
-    cursor             : col-resize;
-    z-index            : 999;
-    border-right       : 2px solid hsla(215, 81%, 79%, .3);    
+    overflow           : hidden;
   }
 `
 
 class App {
   constructor (opts = {}) {
     var self = this
+    self._api = {}
+    var fileStorage = new Storage('sol:')
+    self._api.config = new Config(fileStorage)
+    self._api.filesProviders = {}
+    self._api.filesProviders['browser'] = new Browserfiles(fileStorage)
+    self._api.filesProviders['localhost'] = new SharedFolder(new Remixd())
     self._view = {}
+    self._components = {}
+    self.data = {
+      _layout: {
+        right: {
+          offset: self._api.config.get('right-offset') || 200,
+          show: true
+        }, // @TODO: adapt sizes proportionally to browser window size
+        left: {
+          offset: self._api.config.get('left-offset') || 200,
+          show: true
+        }
+      }
+    }
+    // ----------------- editor ----------------------------
+    self._components.editor = new Editor({}) // @TODO: put into editorpanel
+    // ----------------- editor panel ----------------------
+    self._components.editorpanel = new EditorPanel({
+      api: { editor: self._components.editor },
+      data: { _layout: self.data._layout }
```

Comment:
i see. yeah, i tried a few versions how to best make the "component dance" to synchronize layout when users decide to drag and resize panels.
I'll remove it too